--- query-orig.c	2010-10-11 22:45:55.906022513 +0200
+++ query.c	2010-10-11 22:45:55.909022417 +0200
@@ -582,7 +582,7 @@ YY_LOCAL(int) yyrefill(void)
   while (yybuflen - yypos < 512)
     {
       yybuflen *= 2;
-      yybuf= realloc(yybuf, yybuflen);
+      yybuf= xrealloc(yybuf, yybuflen);
     }
   YY_INPUT((yybuf + yypos), yyn, (yybuflen - yypos));
   if (!yyn) return 0;
@@ -647,7 +647,7 @@ YY_LOCAL(void) yyDo(yyaction action, int
   while (yythunkpos >= yythunkslen)
     {
       yythunkslen *= 2;
-      yythunks= realloc(yythunks, sizeof(yythunk) * yythunkslen);
+      yythunks= xrealloc(yythunks, sizeof(yythunk) * yythunkslen);
     }
   yythunks[yythunkpos].begin=  begin;
   yythunks[yythunkpos].end=    end;
@@ -665,7 +665,7 @@ YY_LOCAL(int) yyText(int begin, int end)
       while (yytextlen < (yyleng - 1))
 	{
 	  yytextlen *= 2;
-	  yytext= realloc(yytext, yytextlen);
+	  yytext= xrealloc(yytext, yytextlen);
 	}
       memcpy(yytext, yybuf + begin, yyleng);
     }
@@ -1203,13 +1203,13 @@ YY_PARSE(int) YYPARSEFROM(yyrule yystart
   if (!yybuflen)
     {
       yybuflen= 1024;
-      yybuf= malloc(yybuflen);
+      yybuf= xmalloc(yybuflen);
       yytextlen= 1024;
-      yytext= malloc(yytextlen);
+      yytext= xmalloc(yytextlen);
       yythunkslen= 32;
-      yythunks= malloc(sizeof(yythunk) * yythunkslen);
+      yythunks= xmalloc(sizeof(yythunk) * yythunkslen);
       yyvalslen= 32;
-      yyvals= malloc(sizeof(YYSTYPE) * yyvalslen);
+      yyvals= xmalloc(sizeof(YYSTYPE) * yyvalslen);
       yybegin= yyend= yypos= yylimit= yythunkpos= 0;
     }
   yybegin= yyend= yypos;
